packages/admin/components/forms/ParentSelector/ParentSelector.tsx

```diff
@@ -0,0 +1,95 @@
+import React from "react";
+import { graphql } from "react-relay";
+import { DialogDisclosure, useDialogState } from "reakit/Dialog";
+import { useTranslation } from "react-i18next";
+import * as Styled from "./ParentSelector.styles";
+import ParentSelectorModal from "./ParentSelectorModal";
+import { useMaybeFragment } from "hooks";
+import BaseInputLabel from "components/forms/BaseInputLabel";
+import { ButtonControl } from "components/atomic";
+
+import { ParentSelectorFragment$key } from "@/relay/ParentSelectorFragment.graphql";
+
+const ParentSelector = ({ data }: Props) => {
+  const entity = useMaybeFragment(fragment, data);
+  const parentFragment =
+    entity?.__typename === "Collection"
+      ? collectionParentFragment
+      : itemParentFragment;
```

Comment:
@1aurend Try fixing the type issue by checking for __typename === "Item", else return null. The error says that `parent` doesn't exist on __typename "%other", so we need to check that the typename is what we expect before getting the parent property.